CONTRIBUTING.md

```diff
@@ -1,52 +1,120 @@
 
-A well formatted issue is appreciated, and goes a long way in helping us help you.
+## Code of Conduct
 
-* Make sure you have a [GitHub account](https://github.com/signup/free)
-* Submit a [GitHub issue](./issues) by:
-  * Clearly describing the issue
-    * Provide a descriptive summary
-    * Explain the expected behavior
-    * Explain the actual behavior
-    * Provide steps to reproduce the actual behavior
-    * Provide your application's complete `Gemfile.lock` as text (in a [Gist](https://gist.github.com) for bonus points)
-    * Any relevant stack traces
+Everyone interacting with Hanami codebases, issue trackers, chat rooms, and forum is expected to follow our [Code of Conduct](./CODE_OF_CONDUCT.md).
 
-If you provide code, make sure it is formatted with the triple backticks (\`).
+## How to contribute
 
-At this point, we'd love to tell you how long it will take for us to respond,
-but we just don't know.
+### Reporting issues
 
-## Pull requests
+- Check [GitHub issues](https://github.com/issues?q=org%3Ahanami+is%3Aopen+is%3Aissue) to see if your issue has already been reported.
+- If you don’t find an open issue, create a new one, either in [hanami/hanami](https://github.com/hanami/hanami/issues) or the [relevant repo](https://github.com/hanami).
+  - Include a clear title and description.
+  - Add as much relevant information as possible (such as your Hanami version,`Gemfile.lock`, Ruby version, OS, as well as code samples, error messages or stack traces).
+  - Include steps or code to reproduce the issue.
 
-We accept pull requests to Hanami for:
+### Fixing issues
 
-* Adding documentation
-* Fixing bugs
-* Adding new features
+We maintain a range of [“help wanted”](https://github.com/issues?q=org%3Ahanami+is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) and [“good first issue”](https://github.com/issues?q=org%3Ahanami+is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) issues on GitHub. These are issues that we want to see addressed for Hanami, and are ready for you to work on.
 
-Not all features proposed will be added but we are open to having a conversation
-about a feature you are championing.
+Feel free to choose an issue that looks interesting to you. Ask questions in the issue if anything is not clear. Once your work on fixing the issue is under way, you can [submit a pull request](#submitting-pull-requests).
 
-Here's a quick guide:
+### Suggesting enhancements
 
-1. Fork the repo.
+If you have a significant enhancement to suggest:
 
-2. Run the tests. This is to make sure your starting point works. Tests can be
-run via `rake`
+- [Create a proposal topic on our forum](https://discourse.hanamirb.org/c/proposals/6) describing your enhancement.
+- Explain why this enhancement might be useful, focusing on real world use cases.
+- Consider how it fits with Hanami framework design, and how it might impact existing features or workflows.
```

Comment:
```suggestion
- Consider how it fits with Hanami's framework design, and how it might impact existing features or workflows.
```

Reply:
This phrasing was intentional :) Thanks for all the other fixes!